CMakeLists.txt

```diff
@@ -278,7 +278,9 @@ if(MSVC)
   if(SDL_CPU_X86)  # don't emit SSE2 in x86 builds
     target_compile_options(SDL2 PRIVATE /arch:SSE)
   endif()
-  set_target_properties(SDL2 PROPERTIES LINK_FLAGS "/NODEFAULTLIB")
+  if(NOT MSVC_CLANG AND NOT SDL_CPU_ARM32)
```

Comment:
Is `MSVC_CLANG` not `clang-cl`? I think we should simply check `MSVC` here??

Reply:
I ported this condition from SDL3's CMakeLists.txt:
```cmake
  # Note: The clang toolset for Visual Studio does not support /NODEFAULTLIB.
  if(MSVC AND NOT SDL_LIBC AND NOT MSVC_CLANG AND NOT SDL_CPU_ARM32)
    # Don't try to link with the default set of libraries.
    if(NOT WINDOWS_STORE)
      # FIXME: is this needed? "-nodefaultlib:MSVCRT" ia already added when SDL_LIBC is false
      target_link_options(SDL3-shared PRIVATE "/NODEFAULTLIB")
    endif()
  endif()
```